Reader: Add border to icon images and featured image

wargcm opened the pull request:
Part of #21645
Ref: p1696503321444809-slack-C05N140C8H5
## Description

Adds a border around the avatar view, site icon view, and featured image view.

## Screenshots

| Light | Dark |
|------|------|
| ![Light](https://github.com/wordpress-mobile/WordPress-iOS/assets/2454408/17029bd1-8076-461f-ae81-4052e17a4435) | ![Screenshot 2023-10-09 at 3 58 56 PM](https://github.com/wordpress-mobile/WordPress-iOS/assets/2454408/a6781b9b-418f-4b07-a83a-e41e266957d0) |

## Testing

To test:
- Launch Jetpack and login
- Enable the feature flag `Reader Improvements v1` if necessary
- Navigate to the Reader
- 🔎 **Verify** There's a border around the image views
- Navigate away and change to either light/dark theme (`CGColor` is not dynamic like `UIColor` so the view needs to be recreated for the correct color)
- Navigate back and 🔎 **verify** there's a border around the image views

## Regression Notes
1. Potential unintended areas of impact
N/A

2. What I did to test those areas of impact (or what existing automated tests I relied on)
N/A

3. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

UI Changes testing checklist:
- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] VoiceOver.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] iPhone and iPad. 
- [ ] Multi-tasking: Split view and Slide over. (iPad)

wargcm committed:
Add border to icon images and featured image

wargcm commented:
@osullivanchris Is the border around the featured image view good here? I matched the colors from your conversation with @dvdchr.

osullivanchris reviewed (approved):
@wargcm hey this looks good so far. So this border resolves separating the image from the background, in the case that they are the same color or very similar. 

There's a separate issue which is separating the two icons from each other in the cases where we have two avatars. My solution there had been to have an additional outside border which matches the background color. I don't see that here. 

Taking light mode example, I was achieving this by:
- having a 10-20% black 0.5pt border inside the image (separates a white image from the white background)
- having a 100% white 1pt border outside the image (provides a gap between two nested images)

One other small issue I've noticed. Sometimes images have transparency. And when there is two nested images, you can 'see through' the first one to the second one, it looks a bit broken. So if possible we could also apply a background color behind the image, matching the background color of the view. In this case, it would be a white circle in light mode behind the user avatar (most of the time it won't be seen, only seen if the avatar has transparency).


Sorry for these super detailed nit picks on something so small. But it will make it feel more polished.

wargcm committed:
Add separator border view

Adds a slightly larger container view for the site icon image view
in order to have a separating border between the avatar image and
the icon image.

It's done this way so that we can have two separate borders. There's
one which gets added to the site icon image view and the second is
added to the new containing view.

wargcm commented:
@osullivanchris Thanks for taking a look!

> There's a separate issue which is separating the two icons from each other in the cases where we have two avatars. My solution there had been to have an additional outside border which matches the background color. I don't see that here.

Yep, I missed that. I added it in https://github.com/wordpress-mobile/WordPress-iOS/pull/21707/commits/e45c1b11c4ffa968dae20ac335e81ea53267d185.

> One other small issue I've noticed. Sometimes images have transparency. And when there is two nested images, you can 'see through' the first one to the second one, it looks a bit broken. So if possible we could also apply a background color behind the image, matching the background color of the view. In this case, it would be a white circle in light mode behind the user avatar (most of the time it won't be seen, only seen if the avatar has transparency).

👍 This should be fixed with the changes from this PR.

dvdchr reviewed (approved):
Works as described. LGTM! :shipit: 

> when there is two nested images, you can 'see through' the first one to the second one, it looks a bit broken. So if possible we could also apply a background color behind the image, matching the background color of the view.

Good point @osullivanchris. I'll take note of this for the Reader Detail avatar images.
